x-pack/plugins/security/public/nav_control/nav_control_component.tsx

```diff
@@ -29,7 +29,7 @@ import { UserAvatar, type UserProfileAvatarData } from '@kbn/user-profile-compon
 import { getUserDisplayName, isUserAnonymous } from '../../common/model';
 import { useCurrentUser, useUserProfile } from '../components';
 
-type ContextMenuItem = EuiContextMenuPanelItemDescriptor & { content?: ReactNode };
+type ContextMenuItem = Omit<EuiContextMenuPanelItemDescriptor, 'content'> & { content?: ReactNode };
```

Comment:
Could I get some context on this change? Why is it necessary?

Reply:
Sure! When I updated to @types/react18 there is a `content: string | undefined` prop that appears on `HTMLAttributes` interface. It propagates to `EuiContextMenuPanelItemDescriptor`. So the `content` prop on `ContextMenuItem` merges into something like `content: ReactNode | string | undefined`. 

This confuses the type below 

![Screenshot 2024-09-10 at 13 11 22](https://github.com/user-attachments/assets/be0080e9-fe9b-4daa-88f7-bc55cd123e02)


So the solution I went with is to explicitly exclude `content` that is coming from  `EuiContextMenuPanelItemDescriptor` and keep only our custom `content`

Reply:
Thank you!